File: sdk/python/examples/dns.py
"""DNS operations with the HyperMesh Python SDK.

Run: python examples/dns.py
"""

from hypermesh import HyperMeshClient, NotFoundError


def main() -> None:
    client = HyperMeshClient()

    # List existing DNS records
    dns = client.dns.list()
    print(f"DNS records: {dns.count}")
    for record in dns.records:
        print(f"  {record.name} -> {record.address}")

    # Register a new record
    print("\nRegistering example.hypermesh -> ::1")
    resp = client.dns.register("example.hypermesh", "::1")
    print(f"Register result: {resp}")

    # Resolve the record
    try:
        resolved = client.dns.resolve("example.hypermesh")
        print(f"Resolved: {resolved.name} -> {resolved.address}")
    except NotFoundError:
        print("Record not found (may not be registered yet)")

    # List records after registration
    updated = client.dns.list()
    print(f"\nDNS records after registration: {updated.count}")
    for record in updated.records:
        print(f"  {record.name} -> {record.address}")


if __name__ == "__main__":
    main()

File: sdk/typescript/examples/dns.ts
/**
 * DNS operations with the HyperMesh TypeScript SDK.
 *
 * Run: npx tsx examples/dns.ts
 */
import { HyperMeshClient, HyperMeshError } from "../src/index.js";

async function main() {
  const client = new HyperMeshClient();

  // List existing DNS records
  const dns = await client.dns.list();
  console.log("DNS records:", dns.count);
  for (const record of dns.records) {
    console.log(`  ${record.name} -> ${record.address}`);
  }

  // Register a new record
  console.log("\nRegistering example.hypermesh -> ::1");
  const reg = await client.dns.register("example.hypermesh", "::1");
  console.log("Register result:", reg);

  // Resolve the record
  try {
    const resolved = await client.dns.resolve("example.hypermesh");
    console.log("Resolved:", resolved.name, "->", resolved.address);
  } catch (err) {
    if (err instanceof HyperMeshError) {
      console.log("Resolve failed:", err.message);
    }
  }

  // List records again to confirm registration
  const updated = await client.dns.list();
  console.log("\nDNS records after registration:", updated.count);
  for (const record of updated.records) {
    console.log(`  ${record.name} -> ${record.address}`);
  }
}

main().catch(console.error);
